src/Sarif.UnitTests/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -0,0 +1,157 @@
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.UnitTests;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class InsertOptionalDataVisitorTests : FileDiffingTests
+    {
+        // To rebaseline all test files set this value to true and rerun the testa
+        private static bool s_rebaseline = false;
+
+        private ITestOutputHelper _outputHelper;
+
+        private static string GetTestDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@".\TestData", subdirectory));
+        }
+
+        private static string GetProductTestDataDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@"..\..\..\..\..\\src\Sarif.UnitTests\TestData", subdirectory));
+        }
+
+        public InsertOptionalDataVisitorTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+        }
+
+        [Theory]
+        [InlineData(OptionallyEmittedData.Hashes)]
+        [InlineData(OptionallyEmittedData.TextFiles)]
+        [InlineData(OptionallyEmittedData.RegionSnippets)]
+        [InlineData(OptionallyEmittedData.ComprehensiveRegionProperties)]
+        [InlineData(OptionallyEmittedData.ComprehensiveRegionProperties | OptionallyEmittedData.RegionSnippets | OptionallyEmittedData.TextFiles | OptionallyEmittedData.Hashes)]
+        public void InsertOptionaDataVisitorTests_InsertsOptionalDataForCommonConditions(OptionallyEmittedData optionallyEmittedData)
+        {
+            string testDirectory = GetTestDirectory("InsertOptionalDataVisitor");
+            string inputFileName = "CoreTests";
+            RunTest(testDirectory, inputFileName, optionallyEmittedData);
+        }
+
+        private void RunTest(string testDirectory, string inputFileName, OptionallyEmittedData optionallyEmittedData)
+        {
+            var sb = new StringBuilder();
+
+            string optionsNameSuffix = "_" + NormalizeOptionallyEmittedDataToString(optionallyEmittedData);
+
+            string expectedFileName = inputFileName + optionsNameSuffix + ".sarif";
+            string actualFileName = inputFileName + optionsNameSuffix + ".actual.sarif";
+            inputFileName = inputFileName + ".sarif";
+
+            expectedFileName = Path.Combine(testDirectory, expectedFileName);
+            actualFileName = Path.Combine(testDirectory, actualFileName);
+            inputFileName = Path.Combine(testDirectory, inputFileName);
+
+            File.Exists(inputFileName).Should().BeTrue();
+
+            SarifLog actualLog;
+
+            JsonSerializerSettings settings = new JsonSerializerSettings()
```

Comment:
>JsonSerializerSettings [](start = 50, length = 22)

Future: This particular `JsonSerializerSettings` object is initialized all over the place (and is even given a name (not appropriate for the current file) `SarifTransformerUtilities.JsonSettingsV2`). We should DRY it out in a separate PR.